events/lunch_buddy.js

```diff
@@ -0,0 +1,422 @@
+                    embeds: [generateAreasEmbed()],
+                    components: areasActionRows,
+                });
+
+                // Setup receiving message interactions
+                const areaCollector = areaMessage.createMessageComponentCollector({
+                    filter: areasButtonsFilter,
+                    time: interactionTimeout,
+                    idle: interactionTimeout,
+                });
+
+                areaCollector.on("collect", async (interaction) => {
+                    const interactorId = String(interaction.user.id);
+                    const priorVoteOption = getVoteOption(interactorId, areaVotes);
+                    const newOption = interaction.customId.replace(areaButtonCustomId, "");
+
+                    let newVoteString = ` voted for ${newOption}!`;
+                    let oldVoteString = "";
+
+                    // Checks whether voter has previously cast vote and edits accordingly
+                    if (priorVoteOption) {
+                        if (priorVoteOption === newOption) {
+                            interaction.reply({
+                                content: "You have already voted for this option.",
+                                ephemeral: true,
+                            });
+                            return;
+                        }
+                        // Removes previously cast vote
+                        const location = areaVotes[priorVoteOption].indexOf(interactorId);
+                        areaVotes[priorVoteOption].splice(location, 1);
+
+                        if (newOption === "Remove") {
+                            newVoteString = " removed your vote.";
+                        } else {
+                            oldVoteString = ` removed your vote for ${priorVoteOption} and`;
+                        }
+                    } else if (newOption === "Remove") {
+                        newVoteString = " no vote to remove.";
+                    }
+
+                    const voteString = `You have${oldVoteString}${newVoteString}`;
+
+                    // Appends new vote if cast
+                    if (newOption !== "Remove") {
+                        areaVotes[newOption].push(interactorId);
+                    }
+
+                    interaction.reply({ content: voteString, ephemeral: true });
+                    areaMessage.edit({ embeds: [generateAreasEmbed(areaVotes)] });
+                });
+
+                areaCollector.on("end", async () => {
+                    // Removes options to vote for area
+                    await areaMessage.edit({ components: [] });
+
+                    let areaInfo;
+
+                    // Finds the highest voted option, and randomises for ties
+                    const mostVoted = getMostVoted(areaVotes);
+
+                    if (mostVoted.length !== 1) {
+                        selectedArea = mostVoted[Math.floor(Math.random() * mostVoted.length)];
```

Comment:
If there are only "any" votes, I believe `mostVoted` just returns an empty array? Think we need to check for that

Same issue for `mostVotedLocation` below as well